addon/components/frost-date-picker.js

```diff
@@ -68,7 +94,8 @@ export default Component.extend(EventsProxyMixin, {
     run.scheduleOnce('sync', this, function () {
       const pikadayOptions = {
         field: this.$('input')[0],
-        onSelect: run.bind(this, this._onSelect)
+        onSelect: run.bind(this, this._onSelect),
```

Comment:
The library does not bind these callback functions to events, the functions are referenced at different points in the selection process https://github.com/weareoutman/clockpicker/blob/gh-pages/src/clockpicker.js#L495 and then the function is simply called https://github.com/weareoutman/clockpicker/blob/gh-pages/src/clockpicker.js#L358